sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -504,6 +513,47 @@ def set_share_quota(self, quota, **kwargs):
             return self._client.share.set_properties( # type: ignore
                 timeout=timeout,
                 quota=quota,
+                access_tier=None,
+                lease_access_conditions=access_conditions,
+                cls=return_response_headers,
+                **kwargs)
+        except StorageErrorException as error:
+            process_storage_error(error)
+
+    @distributed_trace
+    def set_share_properties(self, options, **kwargs):
+        # type: (ShareSetPropertiesOptions, Any) ->  Dict[str, Any]
+        """Sets the share properties.
+
+        .. versionadded:: 12.6.0
+
+        :param options:
```

Comment:
We don't need to worry about overload for python, so I think we can list the options as keyword arguments. What do you think @annatisch @kasobol-msft

Reply:
IMO sounds good. But I'd defer this to python experts.